### PR TITLE
Standardize on input/output terminology for Pulsar Functions

### PR DESCRIPTION
**Pull request by lucperkins**
At the moment, two parallel sets of terminology are in use for Pulsar Functions: source vs. sink and input vs. output. We should standardize on input/output terminology because it's more intuitive and doesn't require users to be aware of terms from graph processing (source/sink).

### CODE REVIEW

File: pulsar-functions/instance/src/main/python/contextimpl.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def record_metric(self, metric_name, metric_value):
       self.accumulated_metrics[metric_name] = AccumulatedMetricDatum()
```
Comment (srkukarni) [inline pulsar-functions/instance/src/main/python/contextimpl.py]: we need to also change the context.py to reflect this

Comment (lucperkins) [reply pulsar-functions/instance/src/main/python/contextimpl.py]: Fixed

**Comment by srkukarni** [comment]
retest this please

**Comment by srkukarni** [comment]
retest this please

**Comment by srkukarni** [comment]
retest this please

**Comment by srkukarni** [comment]
retest this please

**Comment by srkukarni** [comment]
retest this please

**Comment by srkukarni** [comment]
retest this please

**Pull request merged**